### ISSUE TITLE
Just a question: Does this lib really support multiple async listeners?

### ISSUE THREAD

**Issue by nawfalhasan** [pedrolamas/DeferredEvents#2]
Please forgive me if this has been documented. I also haven't tried this library. Just looking at the code I found this:

            var tasks = eventHandler.GetInvocationList()
                .OfType<EventHandler<T>>()
                .Select(invocationDelegate =>
                {
                    cancellationToken.ThrowIfCancellationRequested();

                    invocationDelegate(sender, eventArgs);

                    var deferral = eventArgs.GetCurrentDeferralAndReset();

                    return deferral?.WaitForCompletion(cancellationToken) ?? CompletedTask;
                })
                .ToArray();

            return Task.WhenAll(tasks);

So `InvokeAsync` looks like it executes all the listeners. But in the DeferredEventArgs class I saw:

       public EventDeferral GetDeferral()
        {
            lock (_eventDeferralLock)
            {
                return _eventDeferral ?? (_eventDeferral = new EventDeferral());
            }
        }

In other words, a single instance of EventDeferral is used. That means if any of the listener task is complete, the entire `InvokeAsync` completes.

Right?

**Comment by nawfalhasan** [comment]
I am sorry, I just paid attention to:

       internal EventDeferral GetCurrentDeferralAndReset()
        {
            lock (_eventDeferralLock)
            {
                var eventDeferral = _eventDeferral;

                _eventDeferral = null;

                return eventDeferral;
            }
        }

I had missed  that part. I will close this.
